internal/rotate/file_writer.go

```diff
@@ -95,7 +95,12 @@ func (w *FileWriter) Close() (err error) {
 	defer w.Unlock()
 
 	// Rotate before closing
-	if err = w.rotate(); err != nil {
+	if err = w.rotateIfNeeded(); err != nil {
+		return err
+	}
+
+	// Close the file if we did not rotate
+	if err = w.current.Close(); err != nil {
```

Comment:
Can we please limit the scope of `err` here?
```suggestion
	if err := w.rotateIfNeeded(); err != nil {
		return err
	}

	// Close the file if we did not rotate
	if err := w.current.Close(); err != nil {
```